cmd/internal/cli/build_linux.go

```diff
@@ -36,7 +36,7 @@ func run(cmd *cobra.Command, args []string) {
 		os.Exit(1)
 	}
 
-	if remote {
+	if remote || cmd.Flags().Lookup("builder").Changed {
```

Comment:
I can see that this is the change the issue is asking for: setting builder implies --remote.